pylint/checkers/misc.py

```diff
@@ -137,24 +97,47 @@ def _check_encoding(self, lineno, line, file_encoding):
                                       ' bad encoding'.format(file_encoding))
 
     def process_module(self, module):
-        """inspect the source file to find encoding problem or fixmes like
-        notes
-        """
-        if self.config.notes:
-            notes = re.compile(
-                r'#\s*(%s)\b' % "|".join(map(re.escape, self.config.notes)), re.I)
-        else:
-            notes = None
+        """inspect the source file to find encoding problem"""
         if module.file_encoding:
             encoding = module.file_encoding
         else:
             encoding = 'ascii'
 
         with module.stream() as stream:
             for lineno, line in enumerate(stream):
-                line = self._check_encoding(lineno + 1, line, encoding)
-                if line is not None and notes:
-                    self._check_note(notes, lineno + 1, line, module.tolineno)
+                self._check_encoding(lineno + 1, line, encoding)
+
+    def process_tokens(self, tokens):
+        """inspect the source to find fixme problems"""
+        if not self.config.notes:
+            return
+        comments = [token_info for token_info in tokens
```

Comment:
If I recall correctly, `tokens` is generator, so we can just iterate through it instead of creating a temporary list.